PackageExplorer/Utilities/NuGetExtensions.cs

```diff
@@ -0,0 +1,271 @@
+						string.Compare(x.Value.Framework, framework.Key, StringComparison.InvariantCultureIgnoreCase) == 0 &&
+						string.Compare(x.Value.ToShortName(matchIncludeVersion), frameworkIdentifier,
+							StringComparison.InvariantCultureIgnoreCase) == 0)
+					.Select(x => x).ToList();
+
+			if (includeEquivalentFrameworks)
+			{
+				return result.Select(x => x.Key).Concat(result.Select(x => x.Value));
+			}
+
+			return result.Select(x => x.Value);
+		}
+
+		/// <summary>
+		/// Finds the name of the by framework.
+		/// </summary>
+		/// <param name="frameworkMapping">The framework mapping.</param>
+		/// <param name="frameworkName">Name of the framework.</param>
+		/// <returns>NuGetFramework.</returns>
+		public static NuGetFramework FindByFrameworkName(this IFrameworkMappings frameworkMapping, string frameworkName)
+		{
+			var result = frameworkMapping.EquivalentFrameworks.FirstOrDefault(x =>
+				string.Compare(x.Key.Framework, frameworkName, StringComparison.InvariantCultureIgnoreCase) == 0 ||
+				string.Compare(x.Value.Framework, frameworkName, StringComparison.InvariantCultureIgnoreCase) == 0);
+
+			return result.Key;
+		}
+
+		/// <summary>
+		/// Finds the name based on the full .NET framework name. For Example: WindowsPhone would return the first matching framework in the collection with that name
+		/// </summary>
+		/// <param name="frameworkMapping">The framework mapping.</param>
+		/// <param name="dotNetframeworkName">Name of the dot netframework.</param>
+		/// <returns>NuGetFramework.</returns>
+		public static NuGetFramework FindByDotNetFrameworkName(this IFrameworkMappings frameworkMapping, string dotNetframeworkName)
+		{
+			var result = frameworkMapping.EquivalentFrameworks.FirstOrDefault(x =>
+				string.Compare(x.Key.DotNetFrameworkName, dotNetframeworkName, StringComparison.InvariantCultureIgnoreCase) == 0 ||
+			string.Compare(x.Value.DotNetFrameworkName, dotNetframeworkName, StringComparison.InvariantCultureIgnoreCase) == 0);
+
+			return result.Key;
+		}
+
+		/// <summary>
+		/// Finds the short name by framework.
+		/// </summary>
+		/// <param name="frameworkMapping">The framework mapping.</param>
+		/// <param name="framework">The framework.</param>
+		/// <returns>System.String.</returns>
+		public static string FindShortNameByFramework(this IFrameworkMappings frameworkMapping,
+			NuGetFramework framework)
+		{
+			var result = frameworkMapping.IdentifierShortNames.FirstOrDefault(x => string.Compare(x.Key, framework.Framework, StringComparison.InvariantCultureIgnoreCase) == 0);
+
+			return result.Value;
+		}
+
+		/// <summary>
+		/// Finds the short name of the by short name identifier.  Example: Searching for sl4 would return the framework for SilverLight, Version=4.0
+		/// </summary>
+		/// <param name="frameworks">The frameworks.</param>
+		/// <param name="identifierShortName">Short name of the identifier.</param>
+		/// <returns>IEnumerable&lt;NuGetFramework&gt;.</returns>
```

Comment:
You can use the `<see` tag instead of `IEnumerable&lt;NuGetFramework&gt;.` here.

But in this Case these `<returns>` won't add anything, so please remove it.